src/library/Text/Text.js

```diff
@@ -0,0 +1,165 @@
+  Text_fontSize_mouse: baseTheme.fontSize_mouse,
+  Text_fontSize_prose: baseTheme.fontSize_prose,
+  Text_fontWeight_h1: baseTheme.fontWeight_extraBold,
+  Text_fontWeight_h2: baseTheme.fontWeight_bold,
+  Text_fontWeight_h3: baseTheme.fontWeight_bold,
+  Text_fontWeight_h4: baseTheme.fontWeight_bold,
+  Text_fontWeight_h5: baseTheme.fontWeight_bold,
+  Text_fontWeight_h6: baseTheme.fontWeight_regular,
+  Text_lineHeight: baseTheme.lineHeight_prose,
+  Text_lineHeight_heading: baseTheme.lineHeight,
+  Text_lineHeight_headingSmall: baseTheme.lineHeight_prose,
+  Text_marginBottom: baseTheme.space_stack_md,
+  Text_marginBottom_heading: baseTheme.space_stack_sm,
+
+  ...baseTheme
+});
+
+const styles = {
+  root: ({
+    align,
+    appearance: propAppearance,
+    color,
+    element,
+    fontWeight,
+    noMargins,
+    theme: baseTheme
+  }) => {
+    let theme = componentTheme(baseTheme);
+    const headingValues = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];
+    const isHeading = headingValues.indexOf(element) !== -1;
+    const appearance = propAppearance
+      ? propAppearance
+      : isHeading ? element : null;
+    const isHeadingAppearance = headingValues.indexOf(appearance) !== -1;
+    const rtl = theme.direction === 'rtl';
+
+    if (isHeadingAppearance && appearance) {
+      theme = {
+        ...theme,
+        Text_color: theme[`Text_color_${appearance}`],
+        Text_fontSize: theme[`Text_fontSize_${appearance}`],
+        Text_fontWeight: theme[`Text_fontWeight_${appearance}`],
+        Text_lineHeight:
+          ['h5', 'h6'].indexOf(appearance) !== -1
+            ? theme.Text_lineHeight_headingSmall
+            : theme.Text_lineHeight_heading,
+        Text_marginBottom: theme.Text_marginBottom_heading
+      };
+    }
+
+    let styles = {
+      color: color || theme.Text_color,
+      fontSize:
+        appearance && !isHeadingAppearance
+          ? theme[`Text_fontSize_${appearance}`]
+          : theme.Text_fontSize,
+      fontWeight: (() => {
+        if (fontWeight && theme[`fontWeight_${fontWeight}`]) {
+          return theme[`fontWeight_${fontWeight}`];
+        } else if (isHeadingAppearance && theme.Text_fontWeight) {
+          return theme.Text_fontWeight;
+        } else {
+          return fontWeight || theme.fontWeight_regular;
+        }
+      })(),
+      lineHeight: theme.Text_lineHeight,
+      // Not normalized because we actually want `##em` as applied value
+      marginBottom: !noMargins ? theme.Text_marginBottom : 0,
+      marginTop: 0,
+      textAlign: (() => {
+        if ((rtl && align == 'start') || (!rtl && align == 'end')) {
+          return 'right';
+        } else if ((rtl && align == 'end') || align == 'start') {
+          return 'left';
+        } else {
+          return align;
+        }
+      })(),
+
+      '&::selection': {
+        backgroundColor: theme.Text_backgroundColor_selected
+      }
+    };
+
+    if (['code', 'kbd', 'samp'].indexOf(element) !== -1) {
```

Comment:
Does this list need to include `pre` as well?  It could maybe also be extracted outside of the function.

Reply:
No, `pre` only implies that whitespace is preserved, not that it is displaying source code.

Reply:
It is usually displayed in a monospace font though.

* https://developer.mozilla.org/en-US/docs/Web/HTML/Element/pre
* https://github.com/necolas/normalize.css/blob/master/normalize.css#L57
* https://github.com/twbs/bootstrap/blob/v4-dev/scss/_reboot.scss#L240